src/wallet/test/coinselector_tests.cpp

```diff
@@ -658,4 +663,47 @@ BOOST_AUTO_TEST_CASE(SelectCoins_test)
     }
 }
 
+BOOST_AUTO_TEST_CASE(waste_test)
+{
+    CoinSet selection;
+    const CAmount fee = CAmount(100);
+    const CAmount change_cost = CAmount(100);
+    const CAmount fee_diff = CAmount(50);
+    const CAmount in_amt = 3 * COIN;
+    const CAmount target = 2 * COIN;
+    const CAmount excess = in_amt - fee * 2 - target;
+
+    // Waste with change is the change cost and difference between fee and long term fee
+    add_coin(1 * COIN, 1, selection, fee, fee - fee_diff);
+    add_coin(2 * COIN, 2, selection, fee, fee - fee_diff);
+    const CAmount waste1 = GetSelectionWaste(selection, change_cost, target);
+    BOOST_CHECK_EQUAL(fee_diff * 2 + change_cost, waste1);
+    selection.clear();
+
+    // Waste without change is the excesss and difference between fee and long term fee
+    add_coin(1 * COIN, 1, selection, fee, fee - fee_diff);
+    add_coin(2 * COIN, 2, selection, fee, fee - fee_diff);
+    BOOST_CHECK_EQUAL(fee_diff * 2 + excess, GetSelectionWaste(selection, 0, target));
+    selection.clear();
+
+    // Waste with change and fee == long term fee is just cost of change
+    add_coin(1 * COIN, 1, selection, fee, fee);
+    add_coin(2 * COIN, 2, selection, fee, fee);
+    BOOST_CHECK_EQUAL(change_cost, GetSelectionWaste(selection, change_cost, target));
+    selection.clear();
+
+    // Waste without change and fee == long term fee is just the excess
+    add_coin(1 * COIN, 1, selection, fee, fee);
+    add_coin(2 * COIN, 2, selection, fee, fee);
+    BOOST_CHECK_EQUAL(excess, GetSelectionWaste(selection, 0, target));
+    selection.clear();
+
+    // Waste will be greater when fee is greater, but long term fee is the same
```

Comment:
Perhaps also add a test where the long_term_fee is greater than the fee.

Reply:
Done